thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/EntityType.java

```diff
@@ -20,7 +29,32 @@ public EntityType(String prefix) {
     this.prefix = prefix;
   }
 
+  /**
+   * Returns the parameterized type as string urn. Attaches values in order. Also unwraps the last element
+   * if provided as a Map or Multimap.
+   *
+   * @param values parameters, in order
+   * @return formatted urn
+   */
   public String formatURN(Object... values) {
+    if (values.length <= 0) {
+      return this.prefix;
+    }
+
+    if (values[values.length - 1] instanceof Map) {
```

Comment:
Why we check only the last element? Is this a special use case?

Reply:
It's an internal utility function for writing tests and pipeline code

Reply:
I see. Could you put that in the comment?

Reply:
see javadoc of the method